airflow/providers/amazon/aws/hooks/ec2.py

```diff
@@ -33,21 +33,90 @@ class EC2Hook(AwsBaseHook):
-        super().__init__(resource_type="ec2", *args, **kwargs)
+        super().__init__(client_type="ec2", *args, **kwargs)
 
-    def get_instance(self, instance_id: str):
+    def stop_instances(self, instance_ids):
         """
-        Get EC2 instance by id and return it.
+        Stop instances with given ids
 
-        :param instance_id: id of the AWS EC2 instance
-        :type instance_id: str
-        :return: Instance object
-        :rtype: ec2.Instance
+        :param instance_ids: List of instance ids to stop
+        :return: Dict with key `StoppingInstances` and value as list of instances being stopped
+        """
+        self.log.info("Stopping instances: %s", instance_ids)
+
+        return self.conn.stop_instances(InstanceIds=instance_ids)
+
+    def start_instances(self, instance_ids):
+        """
+        Start instances with given ids
+
+        :param instance_ids: List of instance ids to start
+        :return: Dict with key `StartingInstances` and value as list of instances being started
+        """
+        self.log.info("Starting instances: %s", instance_ids)
+
+        return self.conn.start_instances(InstanceIds=instance_ids)
+
+    def terminate_instances(self, instance_ids):
+        """
+        Terminate instances with given ids
+
+        :param instance_ids: List of instance ids to terminate
+        :return: Dict with key `TerminatingInstances` and value as list of instances being terminated
+        """
+        self.log.info("Terminating instances: %s", instance_ids)
+
+        return self.conn.terminate_instances(InstanceIds=instance_ids)
+
+    def describe_instances(self, filters=None, instance_ids=None):
+        """
+        Describe EC2 instances, optionally applying filters and selective instance ids
+
+        :param filters: List of filters to specify instances to describe
+        :param instance_ids: List of instance IDs to describe
+        :return: Response from EC2 describe_instances API
+        """
+        filters = [] if filters is None else filters
+        instance_ids = [] if instance_ids is None else instance_ids
+
+        self.log.info("Filters provided: %s", filters)
+        self.log.info("Instance ids provided: %s", instance_ids)
+
+        return self.conn.describe_instances(Filters=filters, InstanceIds=instance_ids)
+
+    def get_instances(self, filters=None, instance_ids=None):
         """
-        return self.conn.Instance(id=instance_id)
+        Get list of instance details, optionally applying filters and selective instance ids
+
+        :param instance_ids: List of ids to get instances for
+        :param filters: List of filters to specify instances to get
+        :return: List of instances
+        """
+        description = self.describe_instances(filters=filters, instance_ids=instance_ids)
+
+        return [
+            instance
+            for reservation in description[self.RESERVATIONS] for instance in reservation[self.INSTANCES]
```

Comment:
What about spot instances? I think this would break if the query returned any spot instances.

Reply:
As per [boto3 ec2 describe instance api docs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Client.describe_instances) the response does not change in case of spot instances and `InstanceLifecycle` key indicates whether it's an spot instance or not.
```
InstanceLifecycle (string) --

Indicates whether this is a Spot Instance or a Scheduled Instance.
```